### PR TITLE
proforma/estimate multi currency

### PR DESCRIPTION
**Pull request by ravinderwalkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

proforma/estimate multi currency

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts [hunk 1]
```diff
@@ -148,9 +150,11 @@ export class ProformaListComponent implements OnInit, OnDestroy, OnChanges {
 
     private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
     public isMobileView = false;
+    public baseCurrencySymbol: string = '';
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts:153]: Do add JS doc

File: apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts [hunk 2]
```diff
@@ -167,10 +171,21 @@ export class ProformaListComponent implements OnInit, OnDestroy, OnChanges {
             this.isMobileView = result.matches;
         });
     }
+
     openModal(template: TemplateRef<any>) {
         this.modalRef = this.modalService.show(template);
     }
+
     ngOnInit() {
+        this.store.pipe(select(s => s.settings.profile), takeUntil(this.destroyed$)).subscribe(profile => {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts:180]: Avoid single character variable names

File: apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts [hunk 3]
```diff
@@ -167,10 +171,21 @@ export class ProformaListComponent implements OnInit, OnDestroy, OnChanges {
...
+            if (profile) {
+                this.baseCurrencySymbol = profile.baseCurrencySymbol;
+                this.baseCurrency = profile.baseCurrency;
+            }
+        });
+
+        console.log(this.voucherType);
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts:187]: Avoid console logs

File: apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts [hunk 4]
```diff
@@ -9,31 +9,33 @@ import {
...
+import { InvoiceAdvanceSearchComponent } from '../preview/models/advanceSearch/invoiceAdvanceSearch.component';
+import { GIDDH_DATE_FORMAT } from '../../shared/helpers/defaultDateFormat';
+import { InvoiceSetting } from '../../models/interfaces/invoice.setting.interface';
+import { VoucherTypeEnum } from '../../models/api-models/Sales';
+import { ActivatedRoute, Router } from '@angular/router';
+import { createSelector } from "reselect";
+import { BreakpointObserver } from '@angular/cdk/layout';
+import { GeneralService } from '../../services/general.service';
+
+const MULTI_CURRENCY_MODULES = ['proformas', 'estimates'];
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts:38]: Is this required as the only vouchers that are received to this component are proforma and estimates. Please check

File: apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts [hunk 5]
```diff
@@ -721,9 +740,39 @@ export class ProformaListComponent implements OnInit, OnDestroy, OnChanges {
...
         return obj;
     }
 
+    /**
+     * Adds tooltip text for grand total
+     * to item supplied (for Proforma/Estimate Invoice)
+     *
+     * @private
+     * @param {any} item received from service
+     * @returns {*} Modified item with tooltup text for grand total
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts:754]: Avoid incorrect spelling **tooltup**

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 6]
```diff
@@ -1249,10 +1253,10 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
                 } else {
                     if (result[1]) {
                         result[1] = result[1] as ProformaResponse;
-                        result[1].results.forEach(item => {
+                        result[1].items.forEach(item => {
                             arr.push({
                                 versionNumber: this.isProformaInvoice ? item.proformaNumber : item.estimateNumber,
-                                date: this.isProformaInvoice ? item.proformaDate : item.estimateDate,
+                                date: this.isProformaInvoice ? item.voucherDate : item.voucherDate,
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:1259]: No need of this.isProformaInvoice ?...
```suggestion
                                date: item.voucherDate,
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 7]
```diff
@@ -2935,6 +2958,74 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
         }
         if (this.isProformaInvoice || this.isEstimateInvoice) {
+            let data = requestObject.voucher;
+            let exRate = this.originalExchangeRate;
+            let unqName = this.invoiceUniqueName || this.accountUniqueName;
+
+            let salesEntryClassArray: SalesEntryClassMulticurrency[] = [];
+            let entries = data.entries;
+
+            entries.forEach(e => {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:2968]: Avoid single character variable names

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 8]
```diff
@@ -2935,6 +2958,74 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
+                    let transactionClassMul = new TransactionClassMulticurrency();
+                    transactionClassMul.account.uniqueName = tr.accountUniqueName;
+                    transactionClassMul.account.name = tr.accountName;
+                    transactionClassMul.amount.amountForAccount = tr.amount;
+                    salesEntryClass.hsnNumber = tr.hsnNumber;
+                    salesEntryClass.sacNumber = tr.sacNumber;
+                    salesEntryClass.description = tr.description;
+                    if (tr.isStockTxn) {
+                        let saalesAddBulkStockItems = new SalesAddBulkStockItems();
+                        saalesAddBulkStockItems.name = tr.stockDetails.name;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:2987]: Avoid incorrect spelling **saalesAddBulkStockItems**

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 9]
```diff
@@ -4274,7 +4365,7 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
      * @memberof ProformaInvoiceComponent
      */
     private isMultiCurrencyModule(): boolean {
-        return [VoucherTypeEnum.sales, VoucherTypeEnum.creditNote, VoucherTypeEnum.debitNote, VoucherTypeEnum.cash].includes(this.invoiceType);
+        return [VoucherTypeEnum.sales, VoucherTypeEnum.creditNote, VoucherTypeEnum.debitNote, VoucherTypeEnum.cash, 'proformas', 'estimates'].includes(this.invoiceType);
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:4368]: Do use enums

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 10]
```diff
@@ -2935,6 +2958,74 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
+
+                salesEntryClassArray.push(salesEntryClass);
+            });
+
+            requestObject = {
+                account: data.accountDetails,
+                updateAccountDetails: this.updateAccount,
+                entries: salesEntryClassArray,
+                date: this.convertDateForAPI(data.voucherDetails.voucherDate),
+                type: "sales",
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:3012]: Do use enum

File: apps/web-giddh/src/app/store/proforma/proforma.reducer.ts [hunk 11]
```diff
@@ -144,24 +144,18 @@ export function ProformaReducer(state: ProformaState = initialState, action: Cus
...
-							if (result.body.voucher.voucherDetails.voucherType === VoucherTypeEnum.estimate) {
-								if (m.estimateNumber === result.body.voucher.voucherDetails.estimateNumber) {
-									m.grandTotal = result.body.voucher.voucherDetails.grandTotal;
-								}
-							} else {
-								if (m.proformaNumber === result.body.voucher.voucherDetails.proformaNumber) {
-									m.grandTotal = result.body.voucher.voucherDetails.grandTotal;
-								}
-							}
+						results: vouchers.results.map(m => {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/store/proforma/proforma.reducer.ts:155]: Run formatter

File: apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts (hunk 4 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts:38]: I used the same logic which was getting used.

File: apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts (hunk 5 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/invoice/proforma/proforma-list.component.ts:754]: copy/paste :)

**Pull request merged**